benchmark/bench-report/bin/bench-runner.sh

```diff
@@ -664,8 +669,29 @@ done
 
 if test "$RAW" = "0"
 then
-  run_reports "$TARGETS"
-  run_reports "$COMPARISON_REPORTS"
+
+  if [ -n "$TARGETS" ]
+  then
+    targets_reports=$(run_reports "$TARGETS")
+    if [ -n "$targets_reports" ]
+    then
+        echo "$targets_reports"
+    else
+        echo "No reports generated for the following, $TARGETS"
+    fi
+  fi
+
+  if [ -n "$COMPARISON_REPORTS" ]
+  then
+    comparision_reports=$(run_reports "$COMPARISON_REPORTS")
+    if [ -n "$comparision_reports" ] && [ -n "$COMPARISON_REPORTS" ]
+    then
+        echo "$comparision_reports"
+    else
+        echo "No reports generated for the following, $COMPARISON_REPORTS"
+    fi
+  fi
```

Comment:
I don't like buffering all the output. Why do we even need the "No reports generated" message from this script? The user of this script can examine the output themselves. If you want to detect per target output then you can invoke this script supplying only one target at a time.